include/deal.II/numerics/matrix_creator.templates.h

```diff
@@ -632,22 +632,56 @@ namespace MatrixCreator
 
 
     } // namespace AssemblerBoundary
+
+    
+
+    template <typename T>
+    using compress_t = decltype(std::declval<const T>().compress(
+      std::declval<	VectorOperation>()));
+
+
+
+    template <typename T>
+    constexpr bool has_compress = dealii::internal::is_supported_operation<compress_t, T>;
```

Comment:
Let's use `MatrixType` instead of `T` to make clear what kinds of arguments the template expects.